docs/rpc_statsd.md

```diff
@@ -1,143 +1,155 @@
+    Note: When using multiple fuzzer instances with StatsD it is *strongly* recommended to set up `AFL_STATSD_TAGS_FLAVOR` to match your StatsD server. This will allow you to see individual fuzzer performance, detect bad ones, and see the progress of each strategy.
 
-The simplest way of using this feature is to use any metric provider and change the host/port of your StatsD daemon,
-with `AFL_STATSD_HOST` and `AFL_STATSD_PORT`, if required (defaults are `localhost` and port `8125`).
-To get started, here are some instructions with free and open source tools.
-The following setup is based on Prometheus, statsd_exporter and Grafana.
-Grafana here is not mandatory, but gives you some nice graphs and features.
+3. Optional: To set the host and port of your StatsD daemon, set `AFL_STATSD_HOST` and `AFL_STATSD_PORT`. The default values are `localhost` and `8125`.
 
-Depending on your setup and infrastructure, you may want to run these applications not on your fuzzer instances.
-Only one instance of these 3 application is required for all your fuzzers.
+## Installing and setting up StatsD, Prometheus, and Grafana
 
-To simplify everything, we will use Docker and docker-compose.
-Make sure you have them both installed. On most common Linux distributions, it's as simple as:
+The easiest way to install and set up the infrastructure is with Docker and Docker Compose.
 
-```sh
-curl -fsSL https://get.docker.com -o get-docker.sh
-sh get-docker.sh
-```
+Depending on your fuzzing setup and infrastructure, you may not want to run these applications on your fuzzer instances. This setup may be modified before use in a production environment; for example, adding passwords, creating volumes for storage, tweaking the metrics gathering to get host metrics (CPU, RAM, and so on).
 
-Once that's done, we can create the infrastructure.
-Create and move into the directory of your choice. This will store all the configurations files required.
-
-First, create a `docker-compose.yml` containing the following:
-```yml
-version: '3'
-
-networks:
-  statsd-net:
-    driver: bridge
-
-services:
-  prometheus:
-    image: prom/prometheus
-    container_name: prometheus
-    volumes:
-      - ./prometheus.yml:/prometheus.yml
-    command:
-      - '--config.file=/prometheus.yml'
-    restart: unless-stopped
-    ports:
-      - "9090:9090"
-    networks:
-      - statsd-net
-
-  statsd_exporter:
-    image: prom/statsd-exporter
-    container_name: statsd_exporter
-    volumes:
-      - ./statsd_mapping.yml:/statsd_mapping.yml
-    command:
-      - "--statsd.mapping-config=/statsd_mapping.yml"
-    ports:
-      - "9102:9102/tcp"
-      - "8125:9125/udp"
-    networks:
-      - statsd-net
-  
-  grafana:
-    image: grafana/grafana
-    container_name: grafana
-    restart: unless-stopped
-    ports:
-        - "3000:3000"
-    networks:
-      - statsd-net
-```
+For all your fuzzers, only one instance of StatsD, Prometheus, and Grafana is required.
```

Comment:
StatsD messages are sent out by AFL++, it's not a tool you could have an instance of

Reply:
You maybe need a metric collector (agent, remote backend...) to run (either locally or remotely).
In this case, it's the "prometheus exporter" (image: `prom/statsd-exporter` in the `docker-compose.yml`) but yea that's not "StatsD" per se.

Reply:
Suggestion to fix the sentence:
"For all your fuzzing instances, only one instance of Prometheus and Grafana is required."

Do I need to add anything about the exporter?

Reply:
@Edznux can you clarify about llzmb's questions about the exporter? thanks :)

Reply:
he cant login to github at the moment, he wrote to me: "basically I think it's fine to add just something along the line of "statsd_exporter is useful when dealing with Prometheus, as it converts the statsd metrics to prometheus. If you are using a provider that support statsd directly, you can skip this part If you are going to use the provided Prometheus/graphana stack, you'll need the statsd_exporter"  "

Reply:
Thanks for the input!

Step 2 of the tutorial contains `image: prom/statsd-exporter` in the `docker-compose.yml`. So, we only describe one way to do things. My suggestion:

```suggestion
For all your fuzzing instances, only one instance of Prometheus and Grafana is required. The [statsd exporter](https://registry.hub.docker.com/r/prom/statsd-exporter) converts the StatsD metrics to Prometheus. If you are using a provider that supports StatsD directly, you can skip this part of the setup."
```